langchain4j-core/src/main/java/dev/langchain4j/web/search/WebSearchOrganicResult.java

```diff
@@ -0,0 +1,251 @@
+package dev.langchain4j.web.search;
+
+import dev.langchain4j.data.document.Document;
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.segment.TextSegment;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+
+/**
+ * Represents an organic search results are the web pages that are returned by the search engine in response to a search query.
+ * This includes the title, link, content, and metadata of the web page.
+ * <p>
+ * These results are typically ranked by relevance to the search query.
+ * <p>
+ */
+public class WebSearchOrganicResult {
+
+    private final String title;
+    private final String link;
+    private final String content;
+    private final Map<String, String> resultMetadata;
+
+    /**
+     * Constructs a WebSearchOrganicResult object with the given content.
+     *
+     * @param content The content of the search result.
+     */
+    public WebSearchOrganicResult(String content) {
+        this("noTitle", "noLink", content, null);
```

Comment:
Why "noTitle" and "noLink" instead of null?
Also, how come there can be no link?

Reply:
Normally, any web page found must have a source link and a title at a minimum, otherwise it would not be a web site, but for our purpose it is not worth validating it so as not to dirty the response. I'm going to leave them as null.

Reply:
Naming nitpick: I would rather rename `String link` into `URI url`, to be more consistent and strict.

If url and title are mandatory, please reflect that in the constructors:
```java
public WebSearchOrganicResult(URI url, String title) {
    this.url = ensureNotNull(url, "url");
    this.title = ensureNotBlank(title, "title");
    this.content = null;
    this.metadata = null;
}

public WebSearchOrganicResult(URI url, String title, String content) {
    this.url = ensureNotNull(url, "url");
    this.title = ensureNotBlank(title, "title");
    this.content = ensureNotBlank(content, "content");
    this.metadata = null;
}
```